address.md

```diff
@@ -0,0 +1,403 @@
+# Address
+
+A Filecoin address is an identifier that refers to an actor in the Filecoin state. All actors (miner actors, the storage market actor, account actors) have an address. This address encodes information about the network to which an actor belongs, the specific type of address encoding, the address data itself, and a checksum. The goal of this format is to provide a robust address format that is both easy to use and resistant to errors. Much of this specification was extended from lessons learned in both Bitcoin and Ethereum.
+
+#### Design criteria
+
+1. **Identifiable**: The address must be easily identifiable as a Filecoin address.
+2. **Reliable**: Addresses must provide a mechanism for error detection when they might be transmitted outside the network.
+3. **Upgradable**: Addresses must be versioned to permit the introduction of new address formats.
+4. **Compact**: Given the above constraints, addresses must be as short as possible.
+
+#### Design Considerations
+
+Learning from the Bitcoin:
+
+* Base58 needs a lot of space in QR codes, as it cannot use the ''alphanumeric mode''.
+* The mixed case in base58 makes it inconvenient to reliably write down, type on mobile keyboards, or read out loud.
+* Most of the research on error-detecting codes only applies to character-set sizes that are a [prime power](https://en.wikipedia.org/wiki/Prime_power), which 58 is not.
+* Base58 decoding is complicated and relatively slow.
+
+Learning from the Ethereum:
+
+* Hex encoding is inefficient with respect to storage. Representing addresses as Hex increases the address size by 100%.
+* Checksums require mixed cases, resulting in the same issues as above.
+
+## Specification
+
+A Filecoin address is a byte string consisting of:
+* The **letter 'f'** to identify the address as belonging to a Filecoin network.
```

Comment:
It is unclear to me if this `f` exists in addition to the network indicator or if it is a network indicator. The current implementation treats f as a network indicator.